src/main/java/com/bugsnag/ServletSessionTracker.java

```diff
@@ -0,0 +1,36 @@
+package com.bugsnag;
+
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+/**
+ * Singleton which exposes the {@link SessionTracker} to
+ * {@link com.bugsnag.servlet.BugsnagServletRequestListener},
+ * intended for internal use
+ */
+public enum ServletSessionTracker {
```

Comment:
Why an `enum` here?

Reply:
`BugsnagServletRequestListener` can't take the `SessionTracker` as a constructor parameter, so the current solution was to track sessions statically via a singleton which holds a reference to the `SessionTracker`. According to Effective Java Item 3, an enum is the most foolproof way to create a singleton.

I've mentioned to @kattrali that I'm not entirely convinced that this is the right approach, particularly if a user has created multiple `Bugsnag` instances, as it would just track sessions in the most recently created object. I'm not particularly familiar with web.xml/EE so would be interested to hear anyone's thoughts on alternative approaches.